text/0000-layer-metadata-files.md

```diff
@@ -0,0 +1,184 @@
+# Meta
+[meta]: #meta
+- Name: Combine and organize metadata file locations across lifecycle
+- Start Date: 2021-03-11
+- Author(s): [jabrown85](https://github.com/jabrown85) [samj1912](https://github.com/samj1912)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+This is a proposal to change the default locations of several files and directories. Part of the reasoning behind this is to disambiguate metadata files from application or layer files and avoid naming collisions.
+
+# Motivation
+[motivation]: #motivation
+
+Currently all the metadata files live under the `<layers>` folder. This means that application metadata files like `launch.toml`, `build.toml` or `store.toml` share the same directory as layer metadata files. Since the layers could possibly be named `launch`, `build` or `store` this leads to ambiguity, conflicts and/or restrictions on layer names and their associated metadata files that should not exist. This also prevents us from possibly adding new metadata files in the future without breaking backwards compatibility with buildpacks who may have been creating layers with the same name. This proposal seeks to disambiguate special metadata files from free-form layer metadata files.
+
+
+# What it is
+[what-it-is]: #what-it-is
+
+The proposal is as follows -
+
+The current folder structure is -
+
+```
+/cnb (provided by builder image)
+├── buildpacks
+│   └── <buildpacks>
+├── lifecycle
+│   └── <creator/builder/etc>
+├── order.toml
+└── stack.toml
+/layers (configurable with `CNB_LAYERS_DIR` or -layers)
+├── group.toml
```

Comment:
I don't think that it's important to add it to the RFC, but just in case someone thinks differently, I just want to remind that before platform API 0.5, the default paths of the following files were in the working directory:
```
- group.toml
- plan.toml
- analyzed.toml
- project-metadata.toml
- report.toml
```
So if the working directory was set to `layers` and then the `layers` directory was set to a different directory using `-layers`, these files still exist under `layers`.

For more information, please see https://github.com/buildpacks/lifecycle/issues/393.